src/functions/assertions/PesterThrow.ps1

```diff
@@ -124,13 +124,21 @@
         $but = Join-And $buts
         $failureMessage = "Expected an exception$(if($filter) { " with $filter" }) to be thrown,$(Format-Because $Because) but $but. $actualExceptionLine".Trim()
 
-        return [PSCustomObject] @{
+    $ActualValue = $actualExceptionMessage
+    $ExpectedValue = if ($filterOnExceptionType) { "type $(Format-Nicely $ExceptionType)" } else { 'any exception' }
+
+        return [Pester.ShouldResult] @{
             Succeeded      = $false
             FailureMessage = $failureMessage
+            ExpectResult   = @{
+                Actual   = Format-Nicely $ActualValue
+                Expected = Format-Nicely $ExpectedValue
```

Comment:
Many of those calls just reformat a string that is already formatted, or do 2 calls of Format-Nicely separately on the same data. This looks like unnecessary overhead (calling functions is expensive, and Format-Nicely is also pretty expensive), that is done on every assertion, even if no-one will consume the data. Can this be optimized? Should we add a switch to even populate Actual and Expected?

Reply:
Agreed. `FailureMessage` usually has the same formatted values. Please extract to variable so we can reuse here.

> Should we add a switch to even populate Actual and Expected?
Sounds a bit overkill. If we reuse the formatted values as mentioned above then the performance hit is minimal. This is also exclusive to ailed assertions which you'd normally have few of.

Reply:
Honestly this was a bit of laziness on my part, I just wanted to get all of the various assertions refactored so I tried to get them into a uniform format, I can go through and optimize so that `Format-Nicely` is only called once and when a value is present.

I had considered consolidating a lot of this logic to a single function to keep it DRY, but that would require some dotsourcing or other handling that the build system might not be able to accomidate.

Reply:
> but that would require some dotsourcing or other

Just curious, what did you think about? As long as the helper would be placed in one of these, ex. Pester.Utility, you'd be fine

Reply:
Mostly that a lot of the boilerplate around negate, expected, and actual could be consolidated into a common function for about 80% of the cases. I'll leave it for now as I've already fixed them in place but if I have to fix them again I'll definitely consider this option.